### PR TITLE
Update enhancement handbook for code freeze

### PR DESCRIPTION
**Pull request by npolshakova**
#### What type of PR is this:
/kind cleanup
/kind documentation

Issue: https://github.com/kubernetes/sig-release/issues/2321

Updates to docs on how to find PRR reviewer are addressed here: https://github.com/kubernetes/enhancements/pull/4162. The enhancement template will now include a link to the [production readiness process docs](https://github.com/kubernetes/community/blob/master/sig-architecture/production-readiness.md).

#### What this PR does / why we need it:
1.28 Retro action item called for updating enhancement status tracking for code freeze and documentation around Code Freeze: 
- Add documentation for overall process for the day of Code Freeze and delegating tasks to shadows during the code freeze. 
- Add documentation for new addition options added to the “tracked” label (e.g. “at risk for enhancement freeze”, “at risk for code freeze”).
- Remove KEP readme has an `updated detailed test plan section` requirement from check list (this has been required in the past two releases)
- Add clarification to template that enhancements targeting `stable` need to remove feature gates and set the status as `implemented` after code PRs have merged. 
- Clarify when a KEP needs a PRR review (see [discussion](https://github.com/kubernetes/enhancements/pull/4162#issuecomment-1695678307)). Add a note on the scalability question only being required for beta/GA 
- Add info to enhancement template for how to submit a KEP for a production readiness review

### CODE REVIEW

**Review by Priyankasaggu11929 — CHANGES_REQUESTED** [review]
@npolshakova, thanks so much for creating the PR.

I've requested a few changes and left few review comments. Thanks!

File: release-team/role-handbooks/enhancements/README.md [hunk 1]
```diff
@@ -201,6 +200,20 @@ Read https://github.com/kubernetes/sig-release/blob/master/releases/release_phas
...
+    - Update `Enhancement Status` field for this Enhancement in the Enhancement Tracking Board (`tracked for code freeze` or `at risk for code freeze`)
+- Start planning for the next release while assisting the Release Lead with anything relating to analytics or Public Relation planning of the release. Work with the Communications Lead to develop major themes for the official Kubernetes blog post.
+
+**Note**: out-of-tree enhancments i.e enhancments with label `tracked/out-of-tree` applied to them don't have their PRs inside [kubernetes/kubernetes](https://github.com/kubernetes/kubernetes) repository unlike other enhancements, therefore during enhancements-freeze and code-freeze they might not strictly comply with the tracking criteria.
+  Examples: 
+  - https://github.com/kubernetes/enhancements/issues/3203#issuecomment-1421928726 
+  - https://github.com/kubernetes/enhancements/issues/3476#issuecomment-1467208685
+
+#### Code Freeze Party 🎉
+- Schedule a Code Freeze Party on the day of the code freeze. Invite the Enhancement Shadows and post an invite in the #sig-release and #release-enhancements channels.
```
Comment (Priyankasaggu11929) [inline release-team/role-handbooks/enhancements/README.md:215]:
```suggestion
- On the day of the Code Freeze, schedule a "Code Freeze Party". Invite all the Enhancement shadows, Emiratus Advisor, Release Team Lead and Lead Shadows (make participation optional for the invitees due to potential timezone differences.)
  Post the "Code Freeze Party" meeting details in the #sig-release and #release-enhancements Kubernetes slack channels.
```

**Comment by Priyankasaggu11929** [comment]
(following up from the retrospective discussion on 1.28)

@npolshakova, can we include details regarding the following item as well, in this PR?

`Documentation for Code Freeze: the fact that you have to go to k/k, search up KEP number and KEP keywords`

**Comment by Priyankasaggu11929** [comment]
cc: @gracenng @leonardpahlke @Atharva-Shinde (for further feedback)

File: release-team/role-handbooks/enhancements/README.md [hunk 2]
```diff
@@ -201,6 +200,20 @@ Read https://github.com/kubernetes/sig-release/blob/master/releases/release_phas
 
 #### Week of Code Freeze
 
+- Delegate enhancements to shadows to reach out to assigned KEPs a week before code freeze.
```
[the remark this answers is not in the document]
Comment (npolshakova) [reply release-team/role-handbooks/enhancements/README.md:203]: I agree. I think the one week guideline is from the previous timelines but informally we've been reaching out earlier. Let's change it to 2 weeks?

[the remark this answers is not in the document]
Comment (gracenng) [reply release-team/role-handbooks/enhancements/README.md]: agree with 2 weeks

File: release-team/role-handbooks/enhancements/README.md [hunk 3]
```diff
@@ -164,34 +164,72 @@ It is important that this process be followed and documentation remain up-to-dat
...
-#### Week of Enhancements Freeze
+#### PRR Reviews 
+
+The KEP template production readiness questionnaire should be filled out by the KEP authors, and reviewed by the SIG leads. 
+Once the leads are satisfied with both the overall KEP (i.e., it is ready to move to `implementable` state) and the PRR answers,
+the authors request a PRR approval. See [submitting a KEP for production readiness approval](#submitting-a-kep-for-production-readiness-approval) for more details.
+
+When should a KEP owner request for a new PRR?
+- When the KEP is ready to move to `implementable` state from `provisional`.
+- If the KEP is graduating and is missing a section in the README that was not required for earlier stages (e.g. the scalability section is only required for `beta` or `stable`).
```
Comment (Priyankasaggu11929) [inline release-team/role-handbooks/enhancements/README.md:186]: Regarding ^, I've added some info here: https://github.com/kubernetes/enhancements/pull/4162#issuecomment-1695678307

Also, I pinged the PRR Reviewers team to verify/add info. Let's wait until end of the week for feedback. If not, we'll merge this PR with current set of changes and make more changes in a new PR. Thanks!

Regarding ^, I've added more info here: https://github.com/kubernetes/enhancements/pull/4162#issuecomment-1695678307.

Also, pinged the PRR Reviewers team. Let's wait till week-end for feedback. If not, we'll merge this PR with current changes and follow up later. Thanks!

File: release-team/role-handbooks/enhancements/README.md [hunk 4]
```diff
@@ -236,15 +284,15 @@ This enhancement is targeting for stage `<insert-stage-here>` for {current relea
...
-- [ ] KEP status is marked as `implementable` for `latest-milestone: { CURRENT_RELEASE }`
-- [ ] KEP readme has a [updated detailed test plan section](https://github.com/kubernetes/enhancements/blob/master/keps/NNNN-kep-template/README.md?plain=1#L257-L328) filled out
-- [ ] KEP readme has up to date graduation criteria
-- [ ] KEP has a production readiness review that has been completed and merged into k/enhancements.
+- [ ] KEP status is marked as `implementable` for `latest-milestone: { CURRENT_RELEASE }`. KEPs targeting `stable` will need to be marked as `implemented` after code PRs are merged and the feature gates are removed.
+- [ ] KEP readme has up-to-date graduation criteria
+- [ ] KEP has [submitted a production readiness review](https://github.com/kubernetes/community/blob/master/sig-architecture/production-readiness.md#submitting-a-kep-for-production-readiness-approval) request for approval and has a reviewer assigned.
+- [ ] KEP has a production readiness review that has been completed and merged into k/enhancements.
```
Comment (Priyankasaggu11929) [inline release-team/role-handbooks/enhancements/README.md:290]: @npolshakova, same as what I suggested here - https://github.com/kubernetes/enhancements/pull/4162#issuecomment-1695689592

What do you think? :slightly_smiling_face:

File: release-team/role-handbooks/enhancements/README.md [hunk 5]
```diff
@@ -260,14 +308,34 @@ This enhancement is targeting for stage `<insert-stage-here>` for {current relea
 Here’s where this enhancement currently stands:
 
 - [X] KEP readme using the [latest template](https://github.com/kubernetes/enhancements/tree/master/keps/NNNN-kep-template) has been merged into the k/enhancements repo.
-- [X] KEP status is marked as `implementable` for `latest-milestone: { CURRENT_RELEASE }`
-- [X] KEP readme has a [updated detailed test plan section](https://github.com/kubernetes/enhancements/blob/master/keps/NNNN-kep-template/README.md?plain=1#L257-L328) filled out
-- [X] KEP readme has up to date graduation criteria.
+- [X] KEP status is marked as `implementable` for `latest-milestone: { CURRENT_RELEASE }`. KEPs targeting `stable` will need to be marked as `implemented` after code PRs are merged and the feature gates are removed.
+- [X] KEP readme has up-to-date graduation criteria.
+- [x] KEP has [submitted a production readiness review](https://github.com/kubernetes/community/blob/master/sig-architecture/production-readiness.md#submitting-a-kep-for-production-readiness-approval) request for approval and has a reviewer assigned.
 - [X] KEP has a production readiness review that has been completed and merged into k/enhancements.
```
Comment (Priyankasaggu11929) [inline release-team/role-handbooks/enhancements/README.md:314]: same as what I suggested here - https://github.com/kubernetes/enhancements/pull/4162#issuecomment-1695689592

File: release-team/role-handbooks/enhancements/README.md [hunk 6]
```diff
@@ -361,19 +454,29 @@ Refer to the [Enhancement Freeze Communications](#enhancement-freeze-templates)
...
 - If it is being removed due to missing criteria or lack of response after being included in the milestone, set its status to `Removed from Milestone`.
 
 > Feel free to ask the previous enhancements leads about this when in doubt.
 
+##### Release Team Meeting Updates
+
+For the release team meeting, provide a `status` based on the following criteria:
+- A `Red` on our enhancements status means that something major is blocking our work while tracking the enhancements. We hardly have a red status on our enhancements team.
+- A `Yellow` status generally means that we are still in progress of tracking and reaching out to KEPs and owners, or we have some minor blockages but will be good as they will be resolved soon.
+- A `Green` status means that everything looks good and done from enhancements team side, i.e. every KEP on the tracking board has been communicated with, done with the follow-ups, outreaches etc. Everything is A-OK here.
```
Comment (Priyankasaggu11929) [inline release-team/role-handbooks/enhancements/README.md:477]: Thanks for addressing this. :heart:

**Review by Priyankasaggu11929 — CHANGES_REQUESTED** [review]
Thanks for the changes @npolshakova. :slightly_smiling_face: 

I've added a few more review comments.

**Comment by Priyankasaggu11929** [comment]
@npolshakova, thanks so much for keeping the PR description up-to-date. Truly appreciate it! :heart: :pray:

File: release-team/role-handbooks/enhancements/README.md (hunk 4 above)
Comment (npolshakova) [reply release-team/role-handbooks/enhancements/README.md:290]: Yep, we can combine this new one with the existing `KEP has a production readiness review that has been completed and merged into k/enhancements.` and link to the new documentation!

File: release-team/role-handbooks/enhancements/README.md [hunk 7]
```diff
@@ -201,6 +275,22 @@ Read https://github.com/kubernetes/sig-release/blob/master/releases/release_phas
 
 #### Week of Code Freeze
 
+- Delegate enhancements to shadows to reach out to assigned KEPs two weeks before code freeze. 
+  - Evaluate if the enhancement satisfies all the [requirements](https://github.com/kubernetes/sig-release/blob/master/releases/release_phases.md#code-freeze) for inclusion in the current release.
+    - Comment on the Issue with a status updating using one of the [templates from below](#code-freeze-templates)
+    - Update `Enhancement Status` field for this Enhancement in the Enhancement Tracking Board (`tracked for code freeze` or `at risk for code freeze`)
+    - If the code PRs are not tracked in the GitHub issue description, search [kubernetes/kubernetes](https://github.com/kubernetes/kubernetes) for the KEP number or KEP keywords to find the PRs and add them to the GitHub issue description. Validate that you found the correct code prs with the KEP author.
```
Comment (Priyankasaggu11929) [inline release-team/role-handbooks/enhancements/README.md:282]: minor nit:

```suggestion
    - If the code PRs are not tracked in the GitHub issue description, search [kubernetes/kubernetes](https://github.com/kubernetes/kubernetes) for the KEP number or KEP keywords to find the PRs and add them to the GitHub issue description. Validate that you found the correct code PRs with the KEP author. 
```

**Review by Priyankasaggu11929 — COMMENTED** [review]
Thanks for the changes, @npolshakova.

/lgtm
/approve

Let's remove the hold by EOW as per - https://github.com/kubernetes/sig-release/pull/2297#discussion_r1307458174

**Comment by Priyankasaggu11929** [comment]
Thanks again for all the changes, @npolshakova.

Please squash the commits. And I'll add `lgtm` again. :slightly_smiling_face:

**Pull request merged**